src/server/api/trpc.ts

```diff
@@ -26,9 +30,53 @@ import { DbBaseItemRepository } from "~/lib/domains/mapping/infrastructure/base-
  *
  * @see https://trpc.io/docs/server/context
  */
-export const createTRPCContext = async (opts: { headers: Headers }) => {
+
+// Helper to convert NextApiRequest headers to Fetch API Headers
+// This function will also be used by routers, so it's defined here.
+export function convertToHeaders(
+  incomingHeaders: IncomingHttpHeaders,
+): Headers {
+  const headers = new Headers();
+  for (const [key, value] of Object.entries(incomingHeaders)) {
+    if (value) {
+      if (Array.isArray(value)) {
+        value.forEach((v) => headers.append(key, v));
+      } else {
+        headers.append(key, value);
+      }
+    }
+  }
+  return headers;
+}
+
+export const createContext = async (opts: CreateNextContextOptions) => {
```

Comment:
Looks like we did overcomplexify things with this change. I understand that we want to add user and session to the trpc context, but do we need this complex type with the request and the response? This change provoke other changes in src/commons/trpc/server and src/app/trpc/[trpc]/route.ts. Please make sure these changes are necessary or simplify things